### ISSUE TITLE
v2.7.0b1 throws error when using OpenAI SDK

### ISSUE THREAD

**Issue by willbakst** [pydantic/pydantic#9191]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Trying to generate a simple chat completion (as in below code example) throws:

```shell
AttributeError: type object 'Generic' has no attribute '__annotations__'
```

I've confirmed that this does not happen with previous versions >2.0 and <2.7

### Example Code

```Python
from openai import OpenAI


client = OpenAI(api_key="NONE")  # error is thrown before `api_key` is needed
res = client.chat.completions.create(
    model="gpt-3.5-turbo",
    messages=[{"role": "system", "content": "Say this is a test"}],
)
print(res)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.7.0b1
pydantic-core version: 2.18.0
pydantic-core build: profile=release pgo=false
python version: 3.9.16 (main, Apr  2 2023, 22:08:02)  [Clang 14.0.0 (clang-1400.0.29.202)]
platform: macOS-14.2.1-arm64-arm-64bit
related packages: mypy-1.9.0 typing_extensions-4.10.0 fastapi-0.109.2 pydantic-settings-2.2.1
```

**Comment by samuelcolvin** [comment]
Great catch, thank you. We'll look into it.

**Comment by sydney-runkle** [comment]
@willbakst,

Could you please provide the full traceback? Thanks!

**Comment by dmontagu** [comment]
Oh nevermind, I was able to reproduce it once I switched to python 3.9

**Comment by sydney-runkle** [comment]
We've found the issue and will open a PR with a fix shortly! Should be fixed for 2.7 :).

**Comment by willbakst** [comment]
Awesome, thank you! Would love to test a `v2.7.0b2` as well unless you're releasing `2.7` soon and thus no need :)

**Comment by sydney-runkle** [comment]
@willbakst I think this is a small enough fix that we'll just go ahead with 2.7, but perhaps you could test against main to confirm things are working as expected on your end?

**Comment by dmontagu** [comment]
@willbakst I also confirmed I was able to run your snippet and successfully get output (and it even had `content='This is a test.'` 😄) with the change in #9193.
